aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -185,6 +208,40 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+
+    public entry fun rotate_authentication_key_ed25519(account: &signer, signature: vector<u8>, new_public_key: vector<u8>) acquires Account, OriginatingAddress {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr), error::not_found(EACCOUNT));
+        assert!(
+            vector::length(&new_public_key) == 32,
+            error::invalid_argument(EMALFORMED_PUBLIC_KEY)
+        );
+        assert!(
+            vector::length(&signature) == 64,
+            error::invalid_argument(EMALFORMED_PROOF_OF_KNOWLEDGE)
+        );
+
+        let account_resource = borrow_global_mut<Account>(addr);
+        let current_auth_key = create_address(account_resource.authentication_key);
+
+        let proof = RotationProof {
```

Comment:
@davidiw, this struct was supposed to contain the new public key too, no?

Reply:
I guess we're passing in the new public key in the function parameter? why do we need it in the struct?